nrc_ws/src/nrc_nav/src/nrc_drive_dr.py

```diff
@@ -0,0 +1,124 @@
+#! /usr/bin/env python
+
+#import json
+import rospy
+#import serial
+
+import time
+import csv
+from numpy import genfromtxt
+
+from nrc_msgs.msg import DriveCommand
+
+# Global robot objects
+robot_serial = None
+command_pub = None
+
+# global time
+start_time = time.time()
+# time index for trajectory gen instructions
+instruction_index = 0
+# instructions pulled in at the start of code running
+instructions = None
+
+
+
+# def get_drivetrain_status(timer_event):
+#     global robot_serial
+#     global status_pub
+
+#     # Read the status packet
+#     status = robot_serial.read_until()
+
+#     # Load status into JSON object
+#     try:
+#         status_json = json.loads(status)
+
+#         # Package the status into a message
+#         status_msg = DriveStatus()
+
+#         status_msg.device_id = status_json['id']
+#         status_msg.right_speed = status_json['right_vel']
+#         status_msg.left_speed = status_json['left_vel']
+#         status_msg.acceleration = status_json['acceleration']
+#         status_msg.yaw = status_json['yaw']
+
+#         # Publish the status
+#         status_pub.publish(status_msg)
+
+#     # Sometimes a ValueError occurs when you try to loads(). In this situation just try again next time
+#     except ValueError:
+#         pass
+
+
+
+# def send_drivetrain_command(drive_cmd):
+#     global robot_serial
+
+#     # Form the command packet
+#     command = {"heading": drive_cmd.heading, "speed": drive_cmd.speed}
+
+#     # Convert the packet to a string
+#     command_json = json.dumps(command)
+
+#     # Write the command to serial
+#     robot_serial.write(command_json)
+
+
+def generate_drive_command(timer_event):
+    # get local time at every tick to compare to the path-gen (in seconds)
+    local_time = time.ctime(start_time)
```

Comment:
You might have to include a `global start_time` before this code so you can get the global value, but we can confirm this through testing